### ISSUE TITLE
500 Internal Server Error

### ISSUE THREAD

**Issue by ptman** [hacdias/caddy-filemanager#48]
```text
http://*:8080
root init/

markdown /
browse /

filemanager /_fileman {
    show init/
    allow_commands false
}
```

/_fileman/README.md gave 500 as I described in https://github.com/mholt/caddy/issues/1250

**Comment by hacdias** [comment]
Hello @ptman. Try disabling markdown please and tell me what happens

**Comment by ptman** [comment]
works without markdown, but I'd like to use it

**Comment by hacdias** [comment]
The problem here is that the `markdown` plugin is taking the control of the request instead of `filemanager` because of its higher priority. I can't do much about this but give you an workaround: you can use the File Manager in a subdomain, like:

```
http://*:8080 {
    root init/

    markdown /
    browse /
}

http://fileman.domain:8080 {
    filemanager {
        show init/
        allow_commands false
    }
}
```

**Comment by ptman** [comment]
I see. That would not be quite the outcome I had hoped for. Can @mholt weigh in on this? Why can't the markdown plugin let go of the path controlled by filemanager?

**Comment by hacdias** [comment]
But this behavior is strange since I've already moved `filemanager` to have a bigger priority compared to `markdown`. Are you downloading from caddyserver.com or building from source?

**Comment by ptman** [comment]
It is version 0.9.3 downloaded from caddyserver.com.

**Comment by ptman** [comment]
I did not find clear instructions quickly about how to compile from source with filemanager plugin

**Comment by hacdias** [comment]
I still think this is a really strange behavior. It's clear that the problem is a conflict between markdown and filemanager (pinging @mholt). Can you please try to compile the latest version with filemanager?

Check this: https://forum.caddyserver.com/t/building-caddy-with-latest-selected-plugins/947

**Comment by ptman** [comment]
Ok, I got it build. Doesn't seem to work too well, though. No longer get 500 ISE, but /_fileman/ links are to /_fileman/_fileman/file-i-clicked

Also can't seem to create or edit files.

**Comment by mholt** [comment]
My bad, I need to update the build server. (Sorry!)

**Comment by hacdias** [comment]
That is, indeed, a but @ptman (I think). I'll try to see it ASAP.

**Comment by ebudan** [comment]
Just chiming in to note that I'm seeing the same behaviour (doubled route prefix for Filemanager links; unable to save/create). This is looking at commits  
caddy 12fd34991640c711738692c49218ee0f2664d132
filemanager e03fa7eb0c9e6bd2108da2d10e0a73c7adf34dc9  
...and built from sources, with extra plugins filemanager, expires, search.

I'll be happy to provide my Caddyfile and more details if that helps, although it looks like this is already being taken care of.

**Comment by hacdias** [comment]
Hey @ebudan, can you please send the caddyfile?

**Comment by ebudan** [comment]
Certainly, here's what I'm currently working with. Added .txt to make git co-operate, sorry. 
(Comments about Filemanager and Markdown are from earlier, server-built image, not sure if they still apply.)

[Caddyfile.txt](https://github.com/hacdias/caddy-filemanager/files/587947/Caddyfile.txt)

**Comment by ebudan** [comment]
The duplication of the Filemanager base-url seems to be happening at handlers/listing.go:ServeListing(). Changing

    listing, err := file.GetListing(u, i.VirtualPath, c.AbsoluteURL+r.URL.Path)

to

    listing, err := file.GetListing(u, i.VirtualPath, r.URL.Path)

lets me edit again, but I haven't dug into this enough to be sure there are no side effects. 

The create and save problem happens because Caddy is returning a `501 (Not Implemented)` to the PUT request. I'll have to see if I have time to examine this one.

**Comment by hacdias** [comment]
I'll try to check that today. Thanks :)

On Mon, Nov 14, 2016, 2:20 PM ebudan <notifications@github.com> wrote:

The duplication of the Filemanager base-url seems to be happening at
handlers/listing.go:ServeListing(). Changing

listing, err := file.GetListing(u, i.VirtualPath, c.AbsoluteURL+r.URL.Path)

to

listing, err := file.GetListing(u, i.VirtualPath, r.URL.Path)

lets me edit again, but I haven't dug into this enough to be sure there are
no side effects.

The create and save problem happens because Caddy is returning a 501 (Not
Implemented) to the PUT request. I'll have to see if I have time to examine
this one.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/hacdias/caddy-filemanager/issues/48#issuecomment-260346363>,
or mute the thread

**Comment by hacdias** [comment]
I''m going to solve this now 😄 I just solved it, but I want to improve one or two things.

**Comment by hacdias** [comment]
Hello @ptman! The problem of double URLs is fixed. @ebudan what you told me is fixed too! 

For you both, please try it 😄 I'll look into the other issues.

**Comment by ptman** [comment]
so the double URL may be fixed, but now I get 500 when going to /_fileman/:

500 Internal Server Error

Try reloading the page or hitting the back button. If this error persists, it seems that you may have found a bug! Please create an issue at hacdias/caddy-filemanager repository on GitHub with the code below.

template: listing:116:80: executing "listing" at <.Config.FullWebDavUR...>: can't evaluate field FullWebDavURL in type *config.Config

**Comment by ebudan** [comment]
@ptman, is this still your self-compiled server? Check that you remembered to run `go generate` in the caddy-filemanager directory before building the server. FTR, I've been building back and forth, and am not seeing this.

**Comment by ptman** [comment]
you are absolutely right. I updated caddy and caddy-filemanager but forgot about go generate.

Now, however, creating or uploading files results in 404 response to the PUT request

**Comment by ebudan** [comment]
@ptman Confirming 404 on create and upload - will you open a new bug on that?

**Comment by ptman** [comment]
https://github.com/hacdias/caddy-filemanager/issues/51

**Comment by hacdias** [comment]
I'll check that as soon as I get home. Thanks. And sorry for this :(

**Comment by pepa65** [comment]
Looking forward to seeing this fixed in 0.9.4!
